lang/wasm.cc

```diff
@@ -359,6 +358,32 @@ seastar::future<bytes_opt> run_script(context& ctx, const std::vector<data_type>
     }
 }
 
+seastar::future<bytes_opt> run_script(context& ctx, const std::vector<data_type>& arg_types, const std::vector<bytes_opt>& params, data_type return_type, bool allow_null_input) {
+    auto store = wasmtime::Store(ctx.engine_ptr->get());
+    auto [instance, func] = create_instance_and_func(ctx, store);
+    return run_script(ctx, store, instance, func, arg_types, params, return_type, allow_null_input);
+}
+
+seastar::future<bytes_opt> run_script(const cql3::functions::function_name& name, context& ctx, const std::vector<data_type>& arg_types, const std::vector<bytes_opt>& params, data_type return_type, bool allow_null_input) {
+    auto func_cache = ctx.cache;
+    lw_shared_ptr<wasm_instance> func_inst;
+    std::exception_ptr ex;
+    bytes_opt ret;
+    try {
+        func_inst = func_cache->get(name, ctx).get0();
```

Comment:
regarding the comment above: this helper is also only used once, so you could just as well drop this declaration: ctx.cache is actually shorter